### PR TITLE
Fix automatic issue closing

### PR DESCRIPTION
**Pull request by MVrachev**
**Fixes issue #**: None

**Description of the changes being introduced by the pull request**:

This pr aims to fix the issue that there are occurrences
when a pr is merged, but the issues that it mentions
that it fixes are not automatically closed by GitHub.

According to their documentation:
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
`Fixes` is a keyword that should resolve this.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by MVrachev** [comment]
PS: As proof that `Fixes` works on GitHub I noticed in the prs that the last pr by Jussi [mirrors: Make targets\_path and metadata\_path optional](https://github.com/theupdateframework/tuf/pull/1153) where he uses `Fixes` shows that there is 1 issue linked to the pr:
image

On the other prs it doesn't show that.

**Comment by joshuagl** [comment]
I like this change. Not least of all because I often forget to review and close associated issues when merging PRs.

@lukpueh is there a reason we haven't used this auto-closing reference style before now?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Please fill in the fields below to submit a pull request.  The more information
 that is provided, the better.
 
-**Fixes issue #**:
+**Fixes**:
```
Comment (jku) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe leave the '#' in there to suggest this is not just a title but something that can be filled in?
```suggestion
**Fixes #**
```

**Comment by lukpueh** [comment]
> @lukpueh is there a reason we haven't used this auto-closing reference style before now?

I and others have used them in the past, but it was the responsibility of the PR author to use the add the keyword.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (lukpueh) [reply .github/PULL_REQUEST_TEMPLATE.md]: ... and maybe even add a placeholder?
Something like 
```suggestion
**Fixes #<ISSUE NUMBER>**
```

**Comment by MVrachev** [comment]
Addressed your comments and amended @jku and @lukpueh.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Please fill in the fields below to submit a pull request.  The more information
 that is provided, the better.
 
-**Fixes issue #**:
+**Fixes #<ISSUE NUMBER>**:
```
Comment (joshuagl) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
**Fixes**: #<ISSUE NUMBER>
```

I'm not sure whether this is valid syntax for auto-closing issues, but the after the issue number does not make sense. Maybe remove it completely?

Comment (MVrachev) [reply .github/PULL_REQUEST_TEMPLATE.md]: I updated my pr with your suggestion.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Please fill in the fields below to submit a pull request.  The more information
 that is provided, the better.
 
-**Fixes issue #**:
+**Fixes**: #<ISSUE NUMBER>
```
Comment (jku) [inline .github/PULL_REQUEST_TEMPLATE.md]: Have you checked that this variant works?

**Comment by MVrachev** [comment]
Thanks to @jku for being skeptical here!

He asked me if I have tried to close automatically an issue with the last variant we all liked:
`**Fixes**: #<ISSUE NUMBER>`

I decided to have a look and try this in my own private repository.
What I found is that those:
- `**Fixes**: #<ISSUE NUMBER>`
- `**Fixes** #<ISSUE NUMBER>`

**DOESN'T** automatically close the issue

what **DOES** automatically close the issue is to use
`Fixes:  #<ISSUE NUMBER>`

I updated this pr accordingly.

**Comment by lukpueh** [comment]
Why not just use the [documented syntax](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), i.e. without the colon?

**Comment by MVrachev** [comment]
> Why not just use the [documented syntax](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), i.e. without the colon?

It's a personal preference. For me, it looks better with a colon. 
It works with a colon and without a colon:
```
For example, to close an issue numbered 123, you could use the phrase "Closes #123" or "Closes: #123" 
in your pull request description or commit message. 
Once the branch is merged into the default branch, the issue will close.
```
from https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#about-issue-references

Should I remove the colon?

**Comment by lukpueh** [comment]
> from https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#about-issue-references

You are quoting an outdated GitHub Enterprise doc. The one I linked above is the latest and does not mention colons anymore. 

> Should I remove the colon?

I lean towards *yes*, but I wouldn't insist. ;)

**Comment by MVrachev** [comment]
@lukpueh updated the pr with your recommendations and edited both my pr and commit message to reference to the new documentation.

**Review by lukpueh — APPROVED** [review]
🙇

**Pull request merged**